### ISSUE TITLE
Plan to release etcd v3.5.17

### ISSUE THREAD

**Issue by ivanvc** [etcd-io/etcd#18846]
### What would you like to be added?

The etcd [patch release criteria](https://github.com/etcd-io/etcd/blob/main/Documentation/contributor-guide/release.md#patch-release-criteria) has been met for our `release-3.5` stable release branch so we should release `v3.5.17`.

#### The list of commits included since the previous release is: [v3.5.16...release-3.5](https://github.com/etcd-io/etcd/compare/v3.5.16...release-3.5):

- #18579
- #18599
- #18669
- #18687 
- #18802 
- #18784 
- #18799 
- #18807
- #18842
- #18849 
- #18854

Work in progress CHANGELOG is: https://github.com/etcd-io/etcd/blob/main/CHANGELOG/CHANGELOG-3.5.md#v3516-tbc

#### List of pull requests we still need to backport from `main` to `release-3.5` before the patch release is issued:

- ~#17263~
  - ~#18861~
- ~#18816~
  - ~#18829~
- #18604
  - [x] #18876
- #18812
  - [x] #18863

#### Release team

| GitHub handle | Role |
|---|---|
| @jmhbnz | Release Advisor |
| @ivanvc  | Release Lead |
| @ghouscht   | Release Shadow |
| @stackbaek    | Release Shadow |

### Why is this needed?

Regular patch releases are vital to ensure our users have bug-free and secure software.

**Comment by ivanvc** [comment]
As discussed on Slack, it would be an excellent opportunity to do the release in person next week at KubeCon.

**Comment by ivanvc** [comment]
We should update to Go 1.22.9 (#18847)

**Comment by ivanvc** [comment]
I volunteer to be either the release lead or a shadow.

**Comment by ghouscht** [comment]
I volunteer to help as a shadow. However I'm not at KubeCon and the timezone difference to central europe might complicate things. If you find someone else to help, I'd be happy to  renounce.

**Comment by jmhbnz** [comment]
Discussed during sig-etcd triage meeting. Thanks @ghouscht for volunteering as shadow, I will volunteer as release advisor and with @ivanvc as release lead we are all set to go. We are looking at Tuesday next week MST timezone. 

We are thinking early morning on that Tuesday, how would that work for you @ghouscht?

**Comment by stackbaek** [comment]
I'd also like to be added as a shadow as well. Thanks!

**Comment by stackbaek** [comment]
@ivanvc  - I'd like to be a release shadow for 3.5.

**Comment by ghouscht** [comment]
> Discussed during sig-etcd triage meeting. Thanks @ghouscht for volunteering as shadow, I will volunteer as release advisor and with @ivanvc as release lead we are all set to go. We are looking at Tuesday next week MST timezone.
> 
> We are thinking early morning on that Tuesday Utah time, how would that work for you @ghouscht?

Yes, that does work for me 👍🏻

**Comment by ivanvc** [comment]
Tentatively, the release will be done at [9:00 am MST](https://time.is/compare/0900AM_12_Nov_2024_in_MST).

**Comment by ivanvc** [comment]
Can you help me do a pass at issues and PRs that may need backporting since 3.5.16 (September 10th)? 
* https://github.com/etcd-io/etcd/issues?q=is%3Aissue+created%3A%3E2024-09-10
* https://github.com/etcd-io/etcd/pulls?q=is%3Apr+base%3Amain+created%3A%3E2024-09-10

**Comment by chaochn47** [comment]
Hi @ivanvc, is it a good timing to catch the release train for https://github.com/etcd-io/etcd/pull/17263? 

This bug fix seems like an important one we need to get into release branch.

**Comment by ivanvc** [comment]
Hi @chaochn47, it's still a good time if we can confirm that we want to backport that pull request. I may suggest opening the backport PR and getting feedback from @ahrtr / @serathius  / @fuweid in the backport PR to expedite merging it before Tuesday :)

**Comment by ivanvc** [comment]
I swept through issues/PR and added the ones that caught my eye. Please let me know if you think there are more we should consider.

@ahrtr, should we consider #18829 for this release? Or do we want to hold it because of the tests?

**Comment by ahrtr** [comment]
> @ahrtr, should we consider #18829 for this release? Or do we want to hold it because of the tests?

I already approved https://github.com/etcd-io/etcd/pull/18816, we need second approval to merge the PR. Once the PR for main is merged, then we can approve & merge PR for 3.5.

**Comment by ghouscht** [comment]
> Can you help me do a pass at issues and PRs that may need backporting since 3.5.16 (September 10th)?
> 
>     * https://github.com/etcd-io/etcd/issues?q=is%3Aissue+created%3A%3E2024-09-10
> 
>     * https://github.com/etcd-io/etcd/pulls?q=is%3Apr+base%3Amain+created%3A%3E2024-09-10
> 
> 
> I'll also do a pass later this weekend.

Had a look at the PRs/issues and I think you already caught whats needed. I also opened a PR to backport https://github.com/etcd-io/etcd/pull/18812 to 3.5 and assigned it to you: https://github.com/etcd-io/etcd/pull/18863

**Comment by jmhbnz** [comment]
Hey Team - Update, as discussed in person at KubeCon release `3.5.17` will proceed on schedule tomorrow morning MST and the following items will be deferred to `3.5.18` as they are not considered critical release blocking items:
- #17263 
- #18829

**Comment by ghouscht** [comment]
Looks like we need to update the release steps/script. During the release we noticed that https://github.com/etcd-io/etcd/blob/7ab761246cc449ddfc630001fb2caff160eb4ee3/scripts/release.sh#L133
does not exit with an error so the script doesn't work properly

**Comment by ivanvc** [comment]
Thanks, @ghouscht, for the note.

Just clarifying:

> does not exit with an error so the script doesn't work properly.

The script does exit with an error (that's why it halted), but in the release instructions we should add to verify that `gh auth status` has a clean exit code.

We also need to update the documentation with detailed instructions on publishing the GitHub release.

**Comment by ivanvc** [comment]
Thanks, team (@ghouscht, @stackbaek, @jmhbnz), with all follow-up tasks closed and the release in place. I'll go ahead and close this issue :tada:
